### DISCUSSION TITLE
how to return multiple values? Error creating exporter: value_label must be defined for metric with multiple values

### DISCUSSION THREAD

**Discussion by castlesnow** [burningalchemist/sql_exporter#458] [Q&A] [answered]
cat my.collector.yml
metrics:
  - metric_name: aread
     type: gauge
     help: 'a read'
     key_labels: [c]
    values: [totalread,update_time]
    query: |
      select c,totalread,update_time from b

**Comment by burningalchemist** [comment, marked as answer]
Hey @castlesnow,

as suggested in the error, you'd need to define a `value_label` parameter like in the example below:

```
  #
  # Collected from sys.dm_os_sys_memory
  #
  - metric_name: mssql_os_memory
    type: gauge
    help: 'OS physical memory, used and available.'
    value_label: 'state'
    values: [used, available]
    query: |
      SELECT
        (total_physical_memory_kb - available_physical_memory_kb) * 1024 AS used,
        available_physical_memory_kb * 1024 AS available
      FROM sys.dm_os_sys_memory

  - metric_name: mssql_os_page_file
    type: gauge
    help: 'OS page file, used and available.'
    value_label: 'state'
    values: [used, available]
    query: |
      SELECT
        (total_page_file_kb - available_page_file_kb) * 1024 AS used,
        available_page_file_kb * 1024 AS available
      FROM sys.dm_os_sys_memory
```

The thing is that values requested per query should have some kind of consistency, in this case you can group them by a `value_label` as in the example above.

In your case it seems you have two different value types: the counter (for `totalread`) and the timestamps for `update_time`. If you need to two different values which aren't related, I'd recommend having two separate 2 queries.

---
But I believe, in your case, if you want to append gauge value by an original timestamp, I'd suggest to use a dedicated parameter (introduced in 0.14) - `timestamp_value`.

So then it would be like that:
```
...
query: |
  select c,totalread,update_time from b
values: [totalread]
timestamp_value: update_time
...
```

Please, have a look and let me know if you need more help. I also plan to stop working on features (in fact it's mostly feature complete), and dedicate some time to a documentation page with examples and better descriptions, recipes. So it'd be clearer how to use the exporter. 👍
